components/table/ProductTable.tsx

```diff
@@ -1,104 +1,108 @@
 import React, { useCallback, useEffect, useState } from 'react';
 import { useIntl } from 'react-intl';
 import InfiniteScroll from 'react-infinite-scroll-component';
+import isEqual from 'lodash/isEqual';
 import { getSoftwaresData, getSoftwareListCount } from '../../service/serviceAPI';
 import { ProductsType } from '../../service/types';
 import InfiniteScrollCustomLoader from '../InfiniteScrollLoader';
-import ProductTableHeader from './ProductTableHeader';
 import ProductTableRow from './ProductTableRow';
 import TableErrorHandling from './TableErrorHandling';
 import { SoftwaresTableSortByType } from './types';
+import ProductTableHeader from './ProductTableHeader';
+
+const defaultSortBy = {
+  testApp: { field: 'testApp', order: null },
+  lastUpdate: { field: 'lastUpdate', order: null },
+  overallCompatibility: { field: 'overallCompatibility', order: null }
+} as SoftwaresTableSortByType;
 
 const ProductTable = () => {
   const [productsList, setProductsList] = useState<ProductsType[]>([]);
   const [productListLength, setProductListLength] = useState<number>(0);
-  const [isErrorFetchingData, setIsErrorFetchingData] = useState<boolean>(false);
-  const [isErrorFetchingCount, setIsErrorFetchingCount] = useState<boolean>(false);
-  const [sortBy, setSortBy] = useState<SoftwaresTableSortByType>({
-    testApp: { field: 'testApp', order: null },
-    lastUpdate: { field: 'lastUpdate', order: null },
-    overallCompatibility: { field: 'overallCompatibility', order: null }
-  });
+  const [error, setError] = useState<string | null>(null);
+  const [sortBy, setSortBy] = useState<SoftwaresTableSortByType>(defaultSortBy);
+  const [loading, setLoading] = useState(true);
 
   const { formatMessage } = useIntl();
-  const format = useCallback(
-    (id: string) => formatMessage({ id }),
-    [formatMessage]
-  );
+  const format = useCallback((id: string) => formatMessage({ id }), [formatMessage]);
 
   useEffect(() => {
-    fetchData();
-  }, [sortBy]);
+    const fetchData = async () => {
+      try {
+        const [data, count] = await Promise.all([
+          getSoftwaresData(0, sortBy),
+          getSoftwareListCount()
+        ]);
 
-  const fetchData = async () => {
-    const [data, count] = await Promise.all([
-      getSoftwaresData(0, sortBy),
-      getSoftwareListCount(),
-    ]);
+        if (data.status) {
+          setProductsList(data.data);
+        }
 
-    if (data.status) {
-      setProductsList(data.data);
-    } else {
-      setIsErrorFetchingData(true);
-    }
+        if (count.status) {
+          setProductListLength(count.data);
+        }
+      } catch {
+        setError('Failed to fetch data');
```

Comment:
Is it an internal error or an error presented to the user? If a user can see it then it should be a translation not a hardcoded string

Reply:
It's an internal error used only in scope of ProductTable. If any error occurs then generic TableErorrHandles is displayed. Nevertheless, I've changed the label to camelCase. If in future iteration's we'll be using errors to display something more case specific it'd be good to have it in place.